src/parser/jsonldparser.js

```diff
@@ -0,0 +1,174 @@
+	 * @param {Array} attribute
+	 * @return {Object}
+	 */
+	flatten(context, foldedAttributes, attribute) {
+		let {attributes, things} = foldedAttributes;
+		const {key, value} = attribute;
+
+		if (isObject(value)) {
+			if (this.isEmbbededThing(value)) {
+				const {thing, embeddedThings} = this.parseThing(value);
+				attributes[key] = new Relation(thing.id, thing);
+
+				Object.assign(things, embeddedThings);
+
+				things[thing.id] = thing;
+			} else {
+				const attributesAndThings = this.parseAttributes(
+					value,
+					context
+				);
+				attributes[key] = attributesAndThings.attributes;
+
+				Object.assign(things, attributesAndThings.embeddedThings);
+			}
+		} else if (
+			Array.isArray(value) &&
+			value.length > 0 &&
+			isObject(value[0])
+		) {
+			if (this.isEmbbededThingArray(value)) {
+				const list = value.map(x => this.parseThing(x, things));
+				let relations = [];
+				for (const {thing, embeddedThings} of list) {
+					const relation = new Relation(thing.id, thing);
+					relations.push(relation);
+
+					Object.assign(things, embeddedThings);
+
+					things[thing.id] = thing;
+				}
+
+				attributes[key] = relations;
+			} else {
+				const list = value.map(x => this.parseAttributes(x, context));
+
+				let attributeList = [];
+				for (const attributesAndThings of list) {
+					attributeList.push(attributesAndThings.attributes);
+
+					Object.assign(things, attributesAndThings.embeddedThings);
+				}
+
+				attributes[key] = attributeList;
+			}
+		} else if (this.isId(key, context)) {
+			const relation = new Relation(value, null);
+
+			things[value] = null;
+			attributes[key] = relation;
+		} else {
+			attributes[key] = value;
+		}
+
+		return {attributes, things};
+	}
+
+	/**
+	 * Parses operations from a json array
+	 * @param {Object} json
+	 * @return {Array<Operation>}
+	 */
+	parseOperations(json) {
+		const operationsJson = json['operation'] || [];
+
+		const operations = operationsJson.map(operationJson => {
+			const id = operationJson['@id'];
+			const target = operationJson['target'];
+			const method = operationJson['method'];
+			const expects = operationJson['expects'];
+			const type = operationJson['@type'];
+
+			return new Operation(id, target, method, expects, type);
+		});
+
+		return operations.reduce((acc, operation) => {
+			acc[operation.id] = operation;
+			return acc;
+		}, {});
+	}
+
+	/**
+	 * Check if the object passed is a embbeded thing
+	 * @param {Object} value
+	 * @return {boolean}
+	 */
+	isEmbbededThing(value) {
+		return value['@id'] != null;
+	}
+
+	/**
+	 * Check if the array of object passed if an array of things
+	 * @param {Array<Object>} value
+	 * @return {boolean}
+	 */
+	isEmbbededThingArray(value) {
+		return value[0]['@id'] != null;
+	}
+
+	/**
+	 * Check if the giving name represents a thing id
+	 * @param {String} name
+	 * @param {object} context
+	 * @return {boolean}
+	 */
+	isId(name, context) {
+		if (context) {
+			return (
+				context.filter(
```

Comment:
There is a find

Reply:
It doesn't have support in IE 😭

Reply:
but babel?

Reply:
Same reason than before: babeljs.io/docs/en/babel-plugin-transform-runtime

Reply:
Typescript >>> babel then